### DISCUSSION TITLE
How to use the new API?

### DISCUSSION THREAD

**Discussion by fluppie** [steve-community/steve#1582] [Q&A] [answered]
Hi, I'm running the current github state and I have https working (cost me some fiddling with letsencrypt certs) but it's running.
I tried the swagger API interface but always get:

```
  "status": 401,
  "error": "Unauthorized",
  "message": "The user does not exist, exists but is disabled or has API access disabled.",
```

I click the auth button:
![369839607-85d3bac3-4a47-464d-b9f1-cf77de433861](https://github.com/user-attachments/assets/9c40c9c1-4321-4ec9-afb8-1d345379c23c)
Used the regular admin username and pass and also tried with webapi.key as username and webapi.value as password

Any pointers here on how to get it working :-)?

**Comment by fluppie** [comment]
@goekay @juherr
Sorry for tagging you guys, but I would like to use the API instead of a direct connection to the MariaDB database.
Could you tell me how to get this going?

**Reply by juherr** [reply]
Users are now stored in database.
Check if your user is well configured.

**Comment by fluppie** [comment]
Aha, I think I need to fill-in an API password? In plain text or does it need some encryption?
![afbeelding](https://github.com/user-attachments/assets/182d1851-3c5e-4d52-9b56-4651b26d3689)

**Reply by juherr** [reply]
The `api_password` should be automatically set during the startup phase. See https://github.com/steve-community/steve/pull/1545/files
If not, maybe there is an issue somewhere.

**Reply by MaticBobnar** [reply]
Hi! 
I've noticed the same issue. The api_password is never set and saved during startup phase or read from configuration. 
I solved the issue by copying encoded password to api_password, but that is not really a practical solution.
Should maybe be looked into, I might've missed something

**Reply by fluppie** [reply]
Do you know how to encode? Or can you just copy paste your admin password and use the same, although that's not very smart to re-use it.
Myself I had an earlier Steve git clone (before these changes) and did one or two git pulls to sync to the latest version. Maybe it works fine on a fully new/clean install?
But keen to know how to encode a password so I can fix it manually in the time being :-)

**Reply by MaticBobnar** [reply]
I used copy&paste to test it out if that was indeed the issue - not a smart solution but for testing it's okay.
Afterwards I forked the project and modified the startup phase code so that it reads the api_password property from configuration and sets it accordingly. I set it to use the same encoding method as it is for normal webUser password.
Hope that helps !

**Reply by fluppie** [reply]
Good to know. I'll also try the copy paste thingy.

**Reply by fluppie** [reply]
Copy paste admin's encoded "password" into "api_password" works. But that's of course a dirty workaround :)

**Reply by goekay** [reply, marked as answer]
hey there, i have been waiting https://github.com/steve-community/steve/pull/1544 to stabilize/finalize and be merged to migrate the api pwd to the new flow. since this PR is taking a bit, i will create a fix PR that migrates the existing "header value" as encoded api pwd to the database.

edit: addressed and fixed in https://github.com/steve-community/steve/pull/1608
